packages/block-library/src/image/image.js

```diff
@@ -179,6 +183,23 @@ export default function Image( {
 		}
 	}, [ url, prevUrl ] );
 
+	// Get naturalWidth and naturalHeight from image ref, and fall back to loaded natural
+	// width and height. This resolves an issue in Safari where the loaded natural
+	// witdth and height is otherwise lost when switching between alignments.
+	// See: https://github.com/WordPress/gutenberg/pull/37210.
+	const { naturalWidth, naturalHeight } = useMemo( () => {
+		return {
+			naturalWidth:
+				imageRef.current?.naturalWidth ||
+				loadedNaturalWidth ||
+				undefined,
+			naturalHeight:
+				imageRef.current?.naturalHeight ||
+				loadedNaturalHeight ||
+				undefined,
+		};
+	}, [ loadedNaturalWidth, loadedNaturalHeight, imageRef.current ] );
```

Comment:
I don't think we need to include `imageRef.current` in the dependency, at least based on `exhaustive-deps` 😄

Reply:
Thanks! I've removed the `useMemo` call, since I _think_ setting the `naturalWidth` and `naturalHeight` isn't quite worth it (because there's no real calculation going on). Let me know if you think it's better to leave it in, and I can add it back 🙂

Reply:
Update: I actually just added it back in, as removing it caused the resizer to intermittently fail to display in Safari. I've included `imageRef.current?.complete` in the dependency array — here I _think_ we want the `useMemo` to recalculate _slightly_ more frequently than with just the `loadedNaturalWidth` and `loadedNaturalHeight` values, but happy to remove it if it winds up not being needed.

I've found the behaviour in Safari to be a bit unreliable in testing, so working out what is and isn't needed has been a bit more challenging than usual 😅